optimum/neuron/modeling_decoder.py

```diff
@@ -492,3 +492,10 @@ def push_to_hub(
             revision=revision,
             ignore_patterns=ignore_patterns,
         )
+
+    @staticmethod
```

Comment:
We need to discuss this, as it is a bit hackish.

Reply:
Sure, it comes from this line in transformers: https://github.com/huggingface/transformers/blob/1518508467d96b3866fc4ebcb7a5b3a2e0df2aa4/src/transformers/pipelines/base.py#L883, I did not see an easy workaround other than that, do not want to override the pipeline in transformers neither...

Reply:
I think that's because we don't handle the device correctly

Reply:
Actually we handle it correctly, but no matter what, the pipeline code will call this.
Two options:
- bump transformers to 4.41.0,
- at least check that the device is identical to the hard-coded device we use ("cpu"), and raise an error otherwise.

Reply:
This is fixed in transformers as a side-effect of this pull-request:
https://github.com/huggingface/transformers/pull/30534
The issue actually comes from `optimum` that reimplements the `PretrainedModel` class without providing all its methods (including `to()`).

Reply:
I would suggest the following refactoring to address that issue and potential future issues with missing methods.:
https://github.com/huggingface/optimum-neuron/pull/608
This introduces a common `NeuronModel` class from which all neuron models should inherit.